core/src/main/java/org/infinispan/xsite/OfflineStatus.java

```diff
@@ -158,6 +120,54 @@ public void amend(Integer afterFailures, Long minTimeToWait) {
       amend(builder.create());
    }
 
+   //this is public for testing purpose only
+   public synchronized void updateStatus() {
+      if (firstFailureTime == NO_RECORDING) {
+         return;
+      }
+      internalUpdateStatus();
+   }
+
+   /**
+    * Configures the site to use the supplied configuration for determining when to take a site offline. Also triggers a
+    * state reset.
+    */
+   private void amend(TakeOfflineConfiguration takeOffline) {
+      this.takeOffline = takeOffline;
+      reset();
+   }
+
+   @GuardedBy("this")
+   private void internalUpdateStatus() {
+      if (isOffline) {
+         //already offline
+         return;
+      }
+      boolean hasMinWait = takeOffline.minTimeToWait() > 0;
+      if (hasMinWait && !internalMinTimeHasElapsed()) {
+         //minTimeToWait() not elapsed yet.
+         return;
+      }
+      long minFailures = takeOffline.afterFailures();
+      if (minFailures > 0) {
+         if (minFailures <= failureCount) {
+            if (trace) {
+               log.tracef("Site is failed: min failures (%s) reached (count=%s).", minFailures, failureCount);
+            }
+            listener.siteOffline();
+            isOffline = true;
+         }
+         //else, afterFailures() not reached yet.
+      } else if (hasMinWait) {
+         if (trace) {
+            log.trace("Site is failed: minTimeToWait elapsed and we don't have a min failure number to wait for.");
```

Comment:
This seems wrong, I would expect that if `minFailures` is not configured the site is never taken offline.

The `afterFailures()` javadoc isn't very clear to me:

```
    * The number of failed request operations after which this site should be taken offline. Defaults to 0 (never). A
    * negative value would mean that the site will be taken offline after 'minTimeToWait'.
```

Is that `minTimeToWait` millis from the time the site was brought online? 
Or `minTimeToWait` millis since the first failure (just like with `afterFailures == 1`?

Reply:
both are counted after the first failure and a successful request resets them. 
The code is right. `hasMinWait` is set to `true` so you have minTimeToWait configured and it has elapsed. 
I'll update the javadoc